_posts/2020-02-25-mtc.md

```diff
@@ -0,0 +1,38 @@
+---
+author: Rob Coleman
+comments: false
+date: 2020-02-25
+layout: post
+title: Announcing the MoveIt Task Constructor
+media_type: image
+media_link: /assets/images/blog_posts/mtc/mtc.png
+description: Checkout this new open source feature and tutorial for task planning
+categories:
+- MoveIt
+- ROS
+---
+
+### Introduction
+
+<b>Since its introduction at ROSCon 2018 by <a href="https://github.com/rhaschke" target="_blank">Robert Haschke</a> and <a href="https://github.com/v4hn" target="_blank">Michael Görner</a>, the <a href="https://github.com/ros-planning/moveit_task_constructor" target="_blank">MoveIt Task Constructor (MTC)</a> has gained a lot of interest inside the MoveIt community and in industry. We’re happy to announce that MTC is ready to be used in real-world applications and we present the new official tutorial including a demo application.</b>
+
+The MoveIt Task Constructor (MTC) is a multi-stage planning framework for solving complex motion planning tasks like pick-and-place operations or other object manipulations. With MTC a task is an instance of a high-level planning problem composed of smaller subproblems which are solved by individually stages. These stages are arrangeable components that compute simple steps like moving the robot from a start to a goal state or attaching an object to the gripper. Stages can be freely arranged in sequence or in hierarchical order and even allow arbitrary control flows including alternative and fall-back solutions. The MTC package provides a set of primitive default stages which can already be used to integrate a generic pick-and-place pipeline. Check out the new <a href="https://ros-planning.github.io/moveit_tutorials/doc/moveit_task_constructor/moveit_task_constructor_tutorial.html" target="_blank">tutorial</a> for more information and an example implementation.
```

Comment:
```suggestion
The MoveIt Task Constructor (MTC) is a multi-stage planning framework for solving complex motion planning tasks like pick-and-place operations or other object manipulations. With MTC a task is an instance of a high-level planning problem composed of smaller subproblems that are solved by individual stages. These stages are arrangeable components that compute simple steps like moving the robot between two intermediate states or attaching an object to the gripper. Stages can be freely arranged in sequence or in hierarchical order and even allow arbitrary control flows including alternative and fall-back solutions. The MTC package provides a set of primitive default stages which can already be used to integrate a generic pick-and-place pipeline. Check out the new <a href="https://ros-planning.github.io/moveit_tutorials/doc/moveit_task_constructor/moveit_task_constructor_tutorial.html" target="_blank">tutorial</a> for more information and an example implementation.
```

Reply:
> Actually, @v4hn and me don't consider the MTC API stable yet, which is the major reason we didn't officially release it yet.

@rhaschke 
Would it be okay if we just changed the following sentence?:

From -- "We’re happy to announce that MTC is ready to be used in real-world applications and we present the new official tutorial including a demo application."

To -- "We’re happy to announce that MTC is ready for use and we present the official tutorial including a demo application."

Reply:
Yeah, that sounds more conservative.